cmd/extensions/oidc-policy/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"os"
+
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/kuadrant/kuadrant-operator/cmd/extensions/oidc-policy/internal/controller"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	kuadrantv1 "github.com/kuadrant/kuadrant-operator/api/v1"
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+	extcontroller "github.com/kuadrant/kuadrant-operator/pkg/extension/controller"
+)
+
+var (
+	scheme = k8sruntime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(gatewayapiv1.Install(scheme))
+	utilruntime.Must(kuadrantv1.AddToScheme(scheme))
+	utilruntime.Must(kuadrantv1alpha1.AddToScheme(scheme))
+}
+
+func main() {
+	oidcPolicyReconciler := controller.NewOIDCPolicyReconciler()
+	builder, logger := extcontroller.NewBuilder("oidc-policy-controller")
+	extController, err := builder.
+		WithScheme(scheme).
+		WithReconciler(oidcPolicyReconciler.Reconcile).
+		Watches(&kuadrantv1alpha1.OIDCPolicy{}).
+		Build()
+	if err != nil {
+		logger.Error(err, "unable to create controller")
+		os.Exit(1)
+	}
+
+	oidcPolicyReconciler.SetupWithManager(extController.Manager())
```

Comment:
I need to educate myself in the whole policy extension mechanism, but this smells a lot how we use to do reconciliation in the past and I do not like it. I want to make sure we are not making concurrent calls to a reconcile func that reads again directly from the cluster, that touches so many other resources, based on events centred in the policy object. I bet the next step here is adding mapping functions from target objects to policies ones, and that could be the final breaking of the state of the world principle. Sorry. Not happy with it.

Reply:
This only reads user defined CRs from the cluster, _not_ targets or anything else, those are read from the SotW, through the "kuadrant context", as CEL expression e.g. `self.findGateways()[0].metadata.name` (or whatever it was/is), which is resolved by the Kuadrant controller (and for which then follow up updates are sent to this controller).